dev/NumberBox/InteractionTests/NumberBoxTests.cs

```diff
@@ -0,0 +1,335 @@
+                upButton.InvokeAndWait();
+                Verify.AreEqual(100, numBox.Value);
+
+                Check("WrapCheckBox");
+
+                Log.Comment("Verify that when wrapping is on, clicking the up button wraps to the min value.");
+                upButton.InvokeAndWait();
+                Verify.AreEqual(0, numBox.Value);
+
+                Uncheck("WrapCheckBox");
+
+                Log.Comment("Verify that when wrapping is off, clicking the down button won't go past the min value.");
+                downButton.InvokeAndWait();
+                Verify.AreEqual(0, numBox.Value);
+
+                Check("WrapCheckBox");
+
+                Log.Comment("Verify that when wrapping is on, clicking the down button wraps to the max value.");
+                downButton.InvokeAndWait();
+                Verify.AreEqual(100, numBox.Value);
+            }
+        }
+
+        [TestMethod]
+        public void MinMaxTest()
+        {
+            using (var setup = new TestSetupHelper("NumberBox Tests"))
+            {
+                Check("MinCheckBox");
+                Check("MaxCheckBox");
+
+                RangeValueSpinner numBox = FindElement.ByName<RangeValueSpinner>("TestNumberBox");
+                numBox.SetValue(10);
+                Wait.ForIdle();
+
+                Log.Comment("Verify that setting the value to -1 changes the value to 0");
+                numBox.SetValue(-1);
+                Wait.ForIdle();
+                Verify.AreEqual(0, numBox.Value);
+
+                Log.Comment("Verify that typing '123' in the NumberBox changes the value to 100");
+                EnterText(numBox, "123");
+                Verify.AreEqual(100, numBox.Value);
+
+                Log.Comment("Changing Max to 90; verify value also changes to 90");
+                EnterText(FindElement.ByName<RangeValueSpinner>("MaxNumberBox"), "90");
+                Verify.AreEqual(90, numBox.Value);
+            }
+        }
+
+        [TestMethod]
+        public void BasicCalculationTest()
+        {
+            using (var setup = new TestSetupHelper("NumberBox Tests"))
+            {
+                RangeValueSpinner numBox = FindElement.ByName<RangeValueSpinner>("TestNumberBox");
+
+                Log.Comment("Verify that calculations don't work if AcceptsCalculations is false");
+                EnterText(numBox, "5 + 3");
+                Verify.AreEqual(0, numBox.Value);
+
+                Check("CalculationCheckBox");
+
+                EnterText(numBox, "5 + 3");
+                Verify.AreEqual(8, numBox.Value);
+
+                EnterText(numBox, "9-3*2");
+                Verify.AreEqual(3, numBox.Value);
+
+                EnterText(numBox, "10 /( 2 + 3 )");
+                Verify.AreEqual(2, numBox.Value);
+
+                EnterText(numBox, "2 3 ^");
+                Verify.AreEqual(8, numBox.Value);
+
+                Log.Comment("Verify that diving by zero doesn't work");
```

Comment:
does this raise an error specific to divide by 0? can we check for it.

Reply:
Pinging a forgotten comment

---
In reply to: [341296599](https://github.com/microsoft/microsoft-ui-xaml/pull/1507#discussion_r341296599) [](ancestors = 341296599)